test/query.spec.js

```diff
@@ -193,4 +193,49 @@ describe('query.execute()', () => {
         expect(res.body.results.bindings).toHaveLength(1);
       }));
   });
+
+  describe('update', () => {
+    it('should insert some data', () =>
+      execute(`insert data {:foo :bar :baz}`)
+        .then(res => {
+          expect(res.status).toBe(200);
+          expect(res.body).toBe(null);
+          return execute(`select * {:foo ?p ?o}`);
+        })
+        .then(res => {
+          expect(res.body.results.bindings).toHaveLength(1);
+        }));
+
+    it('should remove some data', () =>
+      execute(`select * {<urn:foo> ?p ?o}`)
+        .then(res => {
+          expect(res.body.results.bindings).toHaveLength(0);
+          return execute('delete data {<urn:foo> <urn:bar> <urn:baz>}');
+        })
+        .then(res => {
+          expect(res.status).toBe(200);
+          return execute('delete data {<urn:foo> <urn:bar> <urn:baz>}');
+        })
+        .then(res => {
+          expect(res.status).toBe(200);
+          return execute(`select * {<urn:foo> ?p ?o}`);
+        })
+        .then(res => {
+          expect(res.body.results.bindings).toHaveLength(0);
+        }));
+
+    it('should work with a delete/insert', () =>
+      execute(`insert {?s :someProp 0} where {?s :bar :baz}`)
+        .then(() =>
+          execute(`
+          delete {?s :someProp ?o}
+          insert {?s :someProp 42}
+          where {?s :someProp ?o}
+        `)
+        )
+        .then(res => {
+          expect(res.status).toBe(200);
```

Comment:
Since we're making assertions that the server did the right thing, we should assert that the {?s :someProp 42} triple is now in the db. Maybe even that the {?s :someProp 0} one isn't.